### PR TITLE
[SDK-2229] Migration guide

### PR DESCRIPTION
**Pull request by adamjmcgrath**
### Description

- Added a migration guide for migrating from 0.x to 1.x
- Couple of fixes on the examples README

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -56,7 +57,7 @@ Take note of the **Client ID**, **Client Secret** and **Domain** of your applica
 
 The library needs the following required configuration keys. These can be configured in a `.env.local` file in the root of your application (See more info about [loading environmental variables in Next.js](https://nextjs.org/docs/basic-features/environment-variables)):
 
-```dotenv
+```sh
```
Comment (adamjmcgrath) [inline README.md:60]: Improves the legibility

File: examples/README.md [hunk 2]
```diff
@@ -4,57 +4,25 @@ In this folder we'll be showing off different examples on how to use the [@auth0
...
+AUTH0_ISSUER_BASE_URL=https://YOUR_AUTH0_DOMAIN
+AUTH0_BASE_URL=http://localhost:3000/
 AUTH0_CLIENT_ID=YOUR_AUTH0_CLIENT_ID
 AUTH0_CLIENT_SECRET=YOUR_AUTH0_CLIENT_SECRET
-REDIRECT_URI=http://localhost:3000/api/callback
-POST_LOGOUT_REDIRECT_URI=http://localhost:3000/
-SESSION_COOKIE_SECRET=viloxyf_z2GW6K4CT-KQD_MoLEA2wqv5jWuq4Jd0P7ymgG5GJGMpvMneXZzhK3sL (at least 32 characters, used to encrypt the cookie)
-```
-
-### Hosting in Zeit Now
```
Comment (adamjmcgrath) [inline examples/README.md:35]: Will get @frederikprijck to add this back once we've sorted out deployments

File: V1_MIGRATION_GUIDE.md [hunk 3]
```diff
@@ -0,0 +1,219 @@
+# V1 Migration Guide
+
+Guide to migrating from `0.x` to `1.x` (Beta)
+
+### Config changes
+
+> Note: If you only use environment variables to configure the SDK you don't need to create an instance of the SDK, you can use the named exports (`handleAuth`, `getSession`) directly from `@auth0/nextjs-auth` and they will lazily create an instance of the SDK for you, and configure it using the following [environment variables](https://auth0.github.io/nextjs-auth0/interfaces/config.config-1.html). See the [Basic setup](./EXAMPLES.md#basic-setup) as an example.
```
Comment (Widcket) [inline V1_MIGRATION_GUIDE.md:7]:
```suggestion
> Note: If you only use environment variables to configure the SDK, you don't need to create an instance of the SDK. You can use the named exports (`handleAuth`, `getSession`) directly from `@auth0/nextjs-auth` and they will lazily create an instance of the SDK for you, and configure it using the following [environment variables](https://auth0.github.io/nextjs-auth0/interfaces/config.config-1.html). See the [Basic setup](./EXAMPLES.md#basic-setup) as an example.
```

File: V1_MIGRATION_GUIDE.md [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+    postLogoutRedirect: '/'
+  },
+  session: {
+    rollingDuration: 60 * 60 * 24,
+    absoluteDuration: 60 * 60 * 24 * 7
+  }
+});
+```
+
+See the API docs for a [full list of configuration options](https://auth0.github.io/nextjs-auth0/interfaces/config.config-1.html)
```
Comment (Widcket) [inline V1_MIGRATION_GUIDE.md:81]:
```suggestion
See the API docs for a [full list of configuration options](https://auth0.github.io/nextjs-auth0/interfaces/config.config-1.html).
```
```suggestion
See the API docs for a [full list of configuration options](https://auth0.github.io/nextjs-auth0/interfaces/config.config-1.html)
```
```suggestion
See the API docs for a [full list of configuration options](https://auth0.github.io/nextjs-auth0/interfaces/config.config-1.html).
```

Comment (Widcket) [inline V1_MIGRATION_GUIDE.md:81]:
```suggestion
See the API docs for a [full list of configuration options](https://auth0.github.io/nextjs-auth0/interfaces/config.config-1.html).
```

File: V1_MIGRATION_GUIDE.md [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+// pages/api/shows.js
+import auth0 from '../lib/auth0';
+
+export default function showsHandler(req, res) {
+  const session = auth0.getSession(req, res); // Note: the extra argument
+  // ...
+}
+```
+
+See the [getSession docs](https://auth0.github.io/nextjs-auth0/modules/session_get_session.html)
```
Comment (Widcket) [inline V1_MIGRATION_GUIDE.md:111]:
```suggestion
See the [getSession docs](https://auth0.github.io/nextjs-auth0/modules/session_get_session.html).
```

File: V1_MIGRATION_GUIDE.md [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+
+### getSession
+
+`getSession` now requires a response as well as a request argument (any updates you make to the session object will now be persisted).
+
+#### Before
+
+```js
+// pages/api/shows.js
+import auth0 from '../lib/auth0';
```
Comment (Widcket) [inline V1_MIGRATION_GUIDE.md:91]:
```suggestion
import auth0 from '../../lib/auth0';
```

File: V1_MIGRATION_GUIDE.md [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+
+export default async function shows(req, res) {
+  const { accessToken } = await auth0.getAccessToken(req, res, {
+    scopes: ['read:shows']
+  });
+  // ...
+}
+```
+
+See the [getAccessToken docs](https://auth0.github.io/nextjs-auth0/modules/session_get_access_token.html)
```
Comment (Widcket) [inline V1_MIGRATION_GUIDE.md:146]:
```suggestion
See the [getAccessToken docs](https://auth0.github.io/nextjs-auth0/modules/session_get_access_token.html).
```

File: V1_MIGRATION_GUIDE.md [hunk 8]
```diff
@@ -0,0 +1,219 @@
...
+
+### handleLogin
+
+The options passed to `handleLogin` have changed
```
Comment (Widcket) [inline V1_MIGRATION_GUIDE.md:150]:
```suggestion
The options passed to `handleLogin` have changed.
```

File: V1_MIGRATION_GUIDE.md [hunk 9]
```diff
@@ -0,0 +1,219 @@
...
+
+- `authParams` is now `authorizationParams`
+- `redirectTo` is now `returnTo`.
```
Comment (Widcket) [inline V1_MIGRATION_GUIDE.md:153]:
```suggestion
- `redirectTo` is now `returnTo`
```

File: V1_MIGRATION_GUIDE.md [hunk 10]
```diff
@@ -0,0 +1,219 @@
...
+      returnTo: '/custom-url'
+    });
+  } catch (error) {
+    console.error(error);
+    res.status(error.status || 400).end(error.message);
+  }
+}
+```
+
+See the [handleLogin docs](https://auth0.github.io/nextjs-auth0/modules/handlers_login.html)
```
Comment (Widcket) [inline V1_MIGRATION_GUIDE.md:203]:
```suggestion
See the [handleLogin docs](https://auth0.github.io/nextjs-auth0/modules/handlers_login.html).
```

File: V1_MIGRATION_GUIDE.md [hunk 11]
```diff
@@ -0,0 +1,219 @@
...
+
+### handleLogout
+
+The options passed to `handleLogout` have changed
```
Comment (Widcket) [inline V1_MIGRATION_GUIDE.md:207]:
```suggestion
The options passed to `handleLogout` have changed.
```

File: V1_MIGRATION_GUIDE.md [hunk 12]
```diff
@@ -0,0 +1,219 @@
...
+
+- `redirectTo` is now `returnTo`.
```
Comment (Widcket) [inline V1_MIGRATION_GUIDE.md:209]:
```suggestion
- `redirectTo` is now `returnTo`
```

File: V1_MIGRATION_GUIDE.md [hunk 13]
```diff
@@ -0,0 +1,219 @@
...
+
+See the [handleLogout docs](https://auth0.github.io/nextjs-auth0/modules/handlers_logout.html)
```
Comment (Widcket) [inline V1_MIGRATION_GUIDE.md:211]:
```suggestion
See the [handleLogout docs](https://auth0.github.io/nextjs-auth0/modules/handlers_logout.html).
```

File: V1_MIGRATION_GUIDE.md [hunk 14]
```diff
@@ -0,0 +1,219 @@
...
+
+### handleCallback
+
+The options passed to `handleCallback` have changed
```
Comment (Widcket) [inline V1_MIGRATION_GUIDE.md:215]:
```suggestion
The options passed to `handleCallback` have changed.
```

File: V1_MIGRATION_GUIDE.md [hunk 15]
```diff
@@ -0,0 +1,219 @@
...
+
+- `onUserLoaded` is now `afterCallback`.
```
Comment (Widcket) [inline V1_MIGRATION_GUIDE.md:217]:
```suggestion
- `onUserLoaded` is now `afterCallback`
```

File: V1_MIGRATION_GUIDE.md [hunk 16]
```diff
@@ -0,0 +1,219 @@
...
+
+See the [handleCallback docs](https://auth0.github.io/nextjs-auth0/modules/handlers_callback.html)
```
Comment (Widcket) [inline V1_MIGRATION_GUIDE.md:219]:
```suggestion
See the [handleCallback docs](https://auth0.github.io/nextjs-auth0/modules/handlers_callback.html).
```

**Review by Widcket — CHANGES_REQUESTED** [review]
Looks good, just some very minor suggestions for clarity and consistency.

**Review by Widcket — APPROVED** [review]

**Pull request merged**